### PR TITLE
Major Refactor

### PR DESCRIPTION
**Pull request by CalmBit**
Wow, this was a pain in the ass. Pretty much redid the entire system to allow for more future extensions, and to clean up a bunch of duplicated code that was creating overhead to adding/modifying materials.

Main Changes:
* Switched from disparate material enums to a singular `EnumMaterial`
* New enum, `EnumProduct`, allows for managing individual material inclusions (blocks/fluids/items)
* Adding a new material consists of creating a new `EnumMaterial` entry.
* Adding a new item simply involves adding a new `EnumProduct` entry followed by item registration in `Smores`.
* Adding a new block does require a subclass, but this is relatively lightweight and doesn't take a lot of time. Same registration code in `Smores`.
* Corrected a bunch of bad bugs that I encountered adding uranium >3<

It's a big one, but I've tried my best to make it as elegant as possible - please do look over it and rip it apart!

### CODE REVIEW

File: src/main/java/com/elytradev/smores/materials/EnumMaterial.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	STEEL("Steel", 			13, 0xFFA4B6BD, 7800, 	1644,	EnumProduct.DEFAULT_ALLOY),
+	BRONZE("Bronze", 		14, 0xFFFC8E00, 8800, 	1186,	EnumProduct.DEFAULT_ALLOY),
+	BRASS("Brass", 			15, 0xFFFCC400, 8520, 	1200,	EnumProduct.DEFAULT_ALLOY),
+	RUBY("Ruby", 			16, 0xFF000000, -1,     -1,		EnumProduct.DEFAULT_GEM),
+	SAPPHIRE("Sapphire", 	17, 0xFF000000, -1,     -1,		EnumProduct.DEFAULT_GEM),
+	PERIDOT("Peridot", 		18, 0xFF000000, -1,     -1,		EnumProduct.DEFAULT_GEM),
+	SULFUR("Sulfur", 		19, 0xFF000000, -1, 	-1,		EnumProduct.DEFAULT_NETHER),
+	NITRE("Nitre", 			20, 0xFF000000, -1,		-1,		EnumProduct.DEFAULT_NETHER),
+	MERCURY("Mercury", 		21, 0xFF000000, -1,		-1,		EnumSet.of(EnumProduct.GLOB)),
+	LAPIS("Lapis", 			22, 0xFF000000, -1,		-1,		EnumSet.of(EnumProduct.DUST));
```
Comment (CalmBit) [inline src/main/java/com/elytradev/smores/materials/EnumMaterial.java]: Wow, the indentation is *fucked* here. I'm going to fix that.

File: src/main/java/com/elytradev/smores/block/BlockGemOre.java [hunk 2]
```diff
@@ -64,55 +53,7 @@ protected BlockStateContainer createBlockState() {
...
-
-	@Override
-	public void getSubBlocks(CreativeTabs tab, NonNullList<ItemStack> items) {
-		int iterator = 0;
-		for (EnumGem gem : EnumGem.values()) {
-			items.add(new ItemStack(this, 1, iterator));
-			iterator++;
-		}
+	public void getDrops(NonNullList<ItemStack> drops, IBlockAccess world, BlockPos pos, IBlockState state, int fortune) {
+		drops.add(new ItemStack(SmoresItems.gem, 1,  getIndexOfMaterial(state.getValue(GEM))));
```
Comment (exaskye) [inline src/main/java/com/elytradev/smores/block/BlockGemOre.java:57]: This won't work with the Fortune enchantment; check the `fortune` parameter (see vanilla BlockOre for the math and random params used)

**Review by exaskye — APPROVED** [review]
LGTM

File: src/main/java/com/elytradev/smores/block/BlockGemOre.java (hunk 2 above)
Comment (CalmBit) [reply src/main/java/com/elytradev/smores/block/BlockGemOre.java:57]: Yeah, this is just temporary until I figure out what I want to have happen with the fortune enchantment >3<

**Review by falkreon — APPROVED** [review]

**Pull request merged**